api/go.mod

```diff
@@ -1,3 +1,3 @@
 module github.com/vanilla-os/vib/api
 
-go 1.21
+go 1.23.0
```

Comment:
Any reason why this is 1.23.0 and the main one is 1.24.0?

Reply:
According to the [documentation](https://go.dev/ref/mod#go-mod-file-go), the `go` directive indicates the minimum version of Go required to compile packages in this module, so it is 1.23.0.

The version in the main `go.mod` is set to the latest version because it can be used in GitHub Actions workflows, to follow the *single source of truth* principle.